### PR TITLE
FI-584: Validate CapabilityStatement

### PR DESCRIPTION
**Pull request by Jammjammjamm**
This branch makes test C-02 validate the server's CapabilityStatement. Before, it just checked that a CapabilityStatement model could be populated from the server's response without actually validating the contents of the CapabilityStatement. 

**Submitter:**
- [x] This pull request describes why these changes were made
- [x] Internal ticket for this PR: https://oncprojectracking.healthit.gov/support/browse/FI-584
- [x] Internal ticket links to this PR
- [x] Internal ticket is properly labeled (Community/Program)
- [x] Internal ticket has a justification for its Community/Program label
- [x] Code diff has been reviewed for extraneous/missing code
- [ ] Tests are included and test edge cases
- [ ] Tests/code quality metrics have been run locally and pass

**Reviewer 1:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

**Reviewer 2:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

File: lib/modules/core/capability_statement_sequence.rb [hunk 1]
```diff
@@ -127,6 +127,10 @@ def json_formats
         rescue StandardError
           assert false, 'Capability Statement could not be parsed.'
         end
+
+        issues = Inferno::RESOURCE_VALIDATOR.validate(@conformance, versioned_resource_class)
```
Comment (okeefm) [inline lib/modules/core/capability_statement_sequence.rb]: Would this make more sense in the `assert_valid_conformance` method? Or is this out of scope for that?

Comment (Jammjammjamm) [reply lib/modules/core/capability_statement_sequence.rb]: I initially tried that, but `assert_valid_conformance` is used in the remaining conformance tests to check that we have received a CapabilityStatement that we can parse. I don't think we want all of those tests failing if we receive an invalid CapabilityStatement. It probably makes the most sense to rename that method to something like `assert_conformance_received`.

Comment (okeefm) [reply lib/modules/core/capability_statement_sequence.rb]: That makes sense to me. Up to you whether you want to rename the method; I'm good approving it as-is.
